### ISSUE TITLE
Show Method Breaks if R Package tictoc is Loaded

### ISSUE THREAD

**Issue by DarioS** [waldronlab/MultiAssayExperiment#321]
Could it be made robust against other packages breaking the display of the experiment list? It shows fine without tictoc loaded.

```r
library(curatedTCGAData)
melanoma <- curatedTCGAData(diseaseCode = "SKCM", assays = c("RNASeq2GeneNorm", "miRNASeqGene"), 
                            version = "2.0.1", dry.run = FALSE)
melanoma
# A MultiAssayExperiment object of 2 listed
# experiments with user-defined names and respective classes.
# Containing an ExperimentList class object of length 2:
# [1] SKCM_miRNASeqGene-20160128: SummarizedExperiment with 1046 rows and 452 columns
# [2] SKCM_RNASeq2GeneNorm-20160128: SummarizedExperiment with 20501 rows and 473 columns
# Functionality:
```
But, loading tictoc prevents the correct display of the experiment list component. It is blank.

```r
library(tictoc)
melanoma
# A MultiAssayExperiment object of 2 listed
# experiments with user-defined names and respective classes.
# Containing an ExperimentList class object of length 2:
# Functionality:
```

**Comment by LiNk-NY** [comment]
Hi Dario, @DarioS 
This is an issue for the `tictoc` maintainers not for `MultiAssayExperiment`. 
Best,
Marcel

**Comment by mtmorgan** [comment]
Can you explain what's going on? It looks like there is `tictoc::as.List`, and this interferes somehow in `vapply()`, where the definition is
```
> vapply
function (X, FUN, FUN.VALUE, ..., USE.NAMES = TRUE)
{
    FUN <- match.fun(FUN)
    if (!is.vector(X) || is.object(X))
        X <- as.list(X)
    .Internal(vapply(X, FUN, FUN.VALUE, USE.NAMES))
}
<bytecode: 0x14a88c3f8>
<environment: namespace:base>
```
and things go wrong
```
> debug(vapply)
> experiments(melanoma)
debugging in: vapply(object, function(o) {
    class(o)[[1L]]
}, character(1L))
debug: {
    FUN <- match.fun(FUN)
    if (!is.vector(X) || is.object(X))
        X <- as.list(X)
    .Internal(vapply(X, FUN, FUN.VALUE, USE.NAMES))
}
Browse[2]> n
debug: FUN <- match.fun(FUN)
Browse[2]>
debug: if (!is.vector(X) || is.object(X)) X <- as.list(X)
Browse[2]>
debug: X <- as.list(X)
Browse[2]> class (X)
[1] "ExperimentList"
attr(,"package")
[1] "MultiAssayExperiment"
Browse[2]> n
debug: .Internal(vapply(X, FUN, FUN.VALUE, USE.NAMES))
Browse[2]> class(X)
[1] "list"
Browse[2]> length(X)
[1] 0
```
but why is `as.list(X)` selecting(?) the wrong method? And can't MultiAssayExperiment (and more generally S4Vectors?) protect against this?

**Comment by LiNk-NY** [comment]
Thanks Martin @mtmorgan for looking into it. 

```r
> conflict_scout()
1 conflict:
* `List`: tictoc, S4Vectors
```

Is there a way to safeguard against the selection of the wrong method in this case? 
Or is it best to avoid `tictoc`?

**Comment by mtmorgan** [comment]
I'm not sure if there's a solution; maybe @hpages has a suggestion.

**Comment by bblodfon** [comment]
Hi @LiNk-NY,

I have also noticed this issue recently (wasn't there 1 year ago) => it can happen even without loading anything else rather than `curatedTCGAData`, `TCGAutils` and `dplyr`! Also, sometimes running via `Rscript` or inside `Rstudio` (even as a background job) leads to different behavior in printing. I haven't been able to narrow it down, I thought it was worth mention it here.

**Comment by hpages** [comment]
I completely missed this, sorry.

Well, in that case we are lucky: List in **S4Vectors** is a _virtual_ class, which means that the class of our objects can never be List. It's always a subclass of List e.g. SimpleList in the case of `S4Vectors::List("a", "b")`. So defining `as.list.SimpleList` addresses the issue, because it takes precedence over any `as.list.List` method.

The fix is in **S4Vectors** 0.37.5 (see https://github.com/Bioconductor/S4Vectors/commit/a0e7cdcc65ba73f1eda79281ba311d611b53c1ed). I also added a few other S3 `as.list` methods.
